### DISCUSSION TITLE
sample workflow json using output of an activity in branching

### DISCUSSION THREAD

**Discussion by mremadevi** [elsa-workflows/elsa-core#5294] [Q&A]
Please provide a sample workflow json using output of an activity in branching?

**Comment by glime-ai** [comment]
Certainly! Below is a sample JSON representation of an Elsa workflow that uses the output of an activity in a branching decision. This example uses a `Switch` activity to branch based on the output of a `ReadLine` activity.

```json
{
  "name": "BranchingExampleWorkflow",
  "variables": [
    {
      "name": "UserChoice",
      "typeName": "String"
    }
  ],
  "activities": [
    {
      "id": "ReadLine1",
      "type": "Elsa.ReadLine",
      "properties": {
        "Output": {
          "expression": {
            "type": "WorkflowVariable",
            "variableName": "UserChoice"
          }
        }
      }
    },
    {
      "id": "Switch1",
      "type": "Elsa.Switch",
      "properties": {
        "Cases": {
          "expression": {
            "type": "JavaScript",
            "value": "{'Approve': 'Branch1', 'Reject': 'Branch2'}"
          }
        },
        "Expression": {
          "expression": {
            "type": "WorkflowVariable",
            "variableName": "UserChoice"
          }
        }
      }
    },
    {
      "id": "Branch1",
      "type": "Elsa.WriteLine",
      "properties": {
        "Text": {
          "expression": {
            "type": "Literal",
            "value": "You chose to Approve."
          }
        }
      }
    },
    {
      "id": "Branch2",
      "type": "Elsa.WriteLine",
      "properties": {
        "Text": {
          "expression": {
            "type": "Literal",
            "value": "You chose to Reject."
          }
        }
      }
    }
  ],
  "connections": [
    {
      "sourceId": "ReadLine1",
      "targetId": "Switch1"
    },
    {
      "sourceId": "Switch1",
      "outcome": "Approve",
      "targetId": "Branch1"
    },
    {
      "sourceId": "Switch1",
      "outcome": "Reject",
      "targetId": "Branch2"
    }
  ]
}
```

In this workflow:

1. The `ReadLine` activity prompts the user for input and stores the result in a variable called `UserChoice`.
2. The `Switch` activity evaluates the value of `UserChoice` and branches to either `Branch1` or `Branch2` based on whether the user input was "Approve" or "Reject".
3. The `WriteLine` activities in `Branch1` and `Branch2` output a message to the console based on the branch taken.

Please note that this is a simplified example and assumes that the `ReadLine` and `WriteLine` activities are part of the Elsa activities library. You may need to adjust the activity types and properties based on the actual activities available in your version of Elsa.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=dRbvTb7IhCSYiI&utm_medium=github).
